resources/upgrading-to-aep/manual-lifecycle-implementation.md

```diff
@@ -0,0 +1,280 @@
+
+{% tabs %}
+
+{% tab title="Android" %}
+#### Java
+
+{% hint style="danger" %}
+Starting and stopping lifecycle should not be done from within a Fragment.
+{% endhint %}
+
+In the `onResume` function of each of your Activities, start Lifecycle data collection:
+
+```java
+@Override  
+public void onResume() {  
+    MobileCore.setApplication(getApplication());
+    MobileCore.lifecycleStart(null);
+}
+```
+
+{% hint style="warning" %}
+To ensure accurate session and crash reporting, this call must be added to every Activity.
+{% endhint %}
+
+{% hint style="info" %}
+Setting the application is only necessary on activities that are entry points for your application. However, setting the application on each Activity has no negative impact and ensures that the SDK always has the necessary reference to your application. We recommend that you call `setApplication` in each of your Activities.
+{% endhint %}
+
+{% endtab %}
+
+{% tab title="iOS (Objective-C)" %}
+#### Objective-C
+
+Start Lifecycle data collection by calling `lifecycleStart:` from within the callback of the `ACPCore::start:` method in your app's `application:didFinishLaunchingWithOptions:` delegate method.
+
+{% hint style="warning" %}
+If your iOS application supports background capabilities, your `application:didFinishLaunchingWithOptions:` method might be called when iOS launches your app in the background. If you do not want background launches to count towards your lifecycle metrics, then `lifecycleStart:` should only be called when the application state is not equal to `UIApplicationStateBackground`.
+{% endhint %}
+
+```objectivec
+- (BOOL) application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // register the lifecycle extension
+    [ACPLifecycle registerExtension];
+
+    const UIApplicationState appState = application.applicationState;
+    [ACPCore start:^{
+        // only start lifecycle if the application is not in the background
+        if (appState != UIApplicationStateBackground) {
+            [ACPCore lifecycleStart:nil];
+        }
+    }];
+}
+```
+
+When launched, if your app is resuming from a backgrounded state, iOS might call your `applicationWillEnterForeground:` delegate method. You also need to call `lifecycleStart:`, but this time you do not need all of the supporting code that you used in `application:didFinishLaunchingWithOptions:`:
+
+```objectivec
+- (void) applicationWillEnterForeground:(UIApplication *)application {
+    [ACPCore lifecycleStart:nil];
+}
+```
+{% endtab %}
+
+{% tab title="iOS (Swift)" %}
+#### Swift
+
+Start Lifecycle data collection by calling `lifecycleStart:` from within the callback of the `ACPCore::start:` method in your app's `application:didFinishLaunchingWithOptions:` delegate method.
```

Comment:
Not using "within" is a pet peeve of mine! Is it still correct to say:

"...from the callback of the..."?